README.md

```diff
@@ -96,7 +78,24 @@ Console.WriteLine("The secret value got from reference is " + resolvedRef);
 ### `ResolveEnv()`
 Take a map of system's environment variables and replaces the values of those which store references of secrets in SecretHub (`secrethub://<path>`) with the value of the respective secret. The other entries in the map remain untouched.
 ```csharp
-Doctionary<string, string> resolvedEnv = client.ResolveEnv(dictionaryToResolve);
+Dictionary<string, string> resolvedEnv = client.ResolveEnv();
+// resolvedEnv = Dictionary<string, string> {
+// 		"REFERENCE_TO_SECRET_ENV_VAR": "some-secret-value",
+//		"NOT_A_REFERENCE_ENV_VAR": "some-other-value"
+// }
+```
+
+### Exceptions
+In order to catch any exceptions thrown by the library (thus avoiding your program to crash), you have to put the lines that use the package in a `try-catch` block.
```

Comment:
I'd except the average reader to be familiar with C# and its exceptions. So I'd say the main goal here is to inform them of the specifics of our exceptions.

```suggestion
Any error encountered encountered by the client, will be thrown as a `ApplicationException`. The full error message can be retrieved from the `Message` field.
```

I'd also consider changing the catch to a `catch(ApplicationException ex)` (is that the correct way to catch such an exception?)

Reply:
Sounds good to me. Adjusted